examples/min-platform/src/main.rs

```diff
@@ -0,0 +1,74 @@
+use anyhow::Result;
+use libloading::os::unix::{Library, Symbol, RTLD_GLOBAL, RTLD_NOW};
+use object::{Object, ObjectSymbol};
+use std::io::Write;
+use std::path::Path;
+
+fn main() -> Result<()> {
+    let target = std::env::args().nth(1).unwrap();
+    let target = Path::new(&target).file_stem().unwrap().to_str().unwrap();
+    // Path to the artifact which is the build of the embedding.
+    //
+    // In this example this is a dynamic library intended to be run on Linux.
+    // Note that this is just an example of an artifact and custom build
+    // processes can produce different kinds of artifacts.
+    let lib = format!("../../target/{target}/release/libembedding.so");
+    let binary = std::fs::read(&lib)?;
+    let object = object::File::parse(&binary[..])?;
+
+    // Showcase verification that the dynamic library in question doesn't depend
+    // on much. Wasmtime build in a "minimal platform" mode is allowed to
+    // depend on some standard C symbols such as `memcpy` but any OS-related
+    // symbol must be prefixed by `wasmtime_*` and be documented in
+    // `crates/runtime/src/sys/custom/capi.rs`.
+    //
+    // This is effectively a double-check of the above assesrtion and showing
+    // how running `libembedding.so` in this case requires only minimal
+    // dependencies.
+    for sym in object.symbols() {
+        if !sym.is_undefined() || sym.is_weak() {
+            continue;
+        }
+
+        match sym.name()? {
+            "" | "memmove" | "memset" | "memcmp" | "memcpy" | "bcmp" | "__tls_get_addr" => {}
```

Comment:
What is the empty symbol used for? Is it for the `STT_FILE` symbols? If so skipping `SymbolKind::File` above makes more sense I think.

Reply:
Ah I didn't question it much, but turns out it's `SymbolKind::Null` so I filtered that out to remove this case.